swap/src/protocol/alice/state.rs

```diff
@@ -403,6 +404,58 @@ impl State3 {
         )
     }
 
+    pub async fn check_for_tx_cancel(
+        &self,
+        bitcoin_wallet: &bitcoin::Wallet,
+    ) -> Result<Transaction> {
+        let tx_cancel = self.tx_cancel();
+        let tx = bitcoin_wallet.get_raw_transaction(tx_cancel.txid()).await?;
+        Ok(tx)
+    }
+
+    pub async fn check_for_tx_refund(
+        &self,
+        bitcoin_wallet: &bitcoin::Wallet,
+    ) -> Result<Transaction> {
+        let tx_refund = self.tx_refund();
+        let tx = bitcoin_wallet.get_raw_transaction(tx_refund.txid()).await?;
+        Ok(tx)
+    }
```

Comment:
This is poorly named. A function named `check_` hints that it doesn't return anything because it performs a check inside but this function doesn't check anything, it just returns you the transaction or an error.